OpenSim/Simulation/Wrap/WrapCylinderObst.h

```diff
@@ -44,29 +43,25 @@ class WrapResult;
  */
 class OSIMSIMULATION_API WrapCylinderObst : public WrapObject {
 OpenSim_DECLARE_CONCRETE_OBJECT(WrapCylinderObst, WrapObject);
-
-//=============================================================================
-// DATA
-//=============================================================================
-
-    enum WrapDirectionEnum  // The prescribed direction of wrapping about the cylinders' z-axis
-    {
-        righthand,
-        lefthand
-    };
-
-    PropertyDbl _radiusProp;
-    double& _radius;
+public:
+//==============================================================================
+// PROPERTIES
+//==============================================================================
+enum WrapDirectionEnum  // The prescribed direction of wrapping about the cylinders' z-axis
+{
+    righthand,
+    lefthand
+};
+
+    OpenSim_DECLARE_PROPERTY(radius, double, "The radius of the cylinder.");
+    OpenSim_DECLARE_PROPERTY(length, double, "The length of the cylinder.");
+    OpenSim_DECLARE_PROPERTY(wrapDirectionName, std::string, "Describe if the cylinder is right or left handed.");
```

Comment:
In the previous code, it defines later wrapDirection in terms of what is in wrapDirectectionName (righthanded when wrapDirectectionName is RightHanded or Undefined and lefthanded when wrapDirectionName is LeftHanded). That is why I kept both. But if there is a way to use directly wrapDirectionEnum like this:
`OpenSim_DECLARE_PROPERTY(wrapDirection, WrapDirectionEnum, "Describe if the cylinder is right or left handed.");`
This is definitely better. Otherwise, we have to keep them separate and initialize WrapDirection when it is created or copied

Reply:
Unfortunately we can't use enums as the type for properties. You'll still need to use a string. I'm just saying that the property name used to be "wrapDirection" (that's the XML tag that would have appeared) and that for backwards compatibility, the first macro argument here should be "wrapDirection".

Reply:
Ok, I see, I'll make the change